File: nodejs/device/intel-edison/examples/button.js
/**
 *   This sample code demo button state send to QIoT Suite Lite by MQTT protocol
 *   requirement:
 *   -- npm install 
 *   run command: node button.js
 */

var qiot = require('./lib/qiot');
var m = require('mraa');
var pre_button_state;

/**
 * Setup Gpio
 * set pin2 is INPUT pin 
 */
var button_pin = 2;
var button_sensor = new m.Gpio(button_pin); //setup digital read on pin 2
button_sensor.dir(m.DIR_IN); //set the gpio direction to ignput

/**
 * Setup connection options
 */
var connection = new qiot(qiot.protocol.MQTT);
var connection_option = connection.readResource('./res/resourceinfo.json', './ssl/');

/**
 * Send sensor's data to QIoT Suite Lite.
 * content of ./res/resourceinfo.json
 * {
 *     ...
 *     "resources": [
 *         {
 *              ...
 *              "resourceid": "temp",
 *              "topic": "qiot/things/admin/abccccc/temp",
 *              ...
 *          }
 *      ]
 *  }
 */
connection.on('connect', function(data) {
    // detect button state every 1000 msec and publish when button state change
    setInterval(function() {
        console.log("senosr read:" + button_sensor.read());
        if (pre_button_state != button_sensor.read()) {
            // TODO: you could replace "button" by any resource id set form QIoT Suite Lite
            connection.publishById("button", button_sensor.read());
            pre_button_state = button_sensor.read();
            // or publish by resource topic
            // TODO: you could replace "qiot/things/admin/edison/button" by any Topic form QIoT Suite Lite like following
            // connection.publishByTopic("qiot/things/admin/edison/button", getRandomInt(0, 50));
        }
    }, 1000);
});
connection.connect(connection_option);

File: python/device/mtk-linkit-7688-duo/examples/button.py
# -*- coding: utf-8 -*-
import time
import sys
from lib import qiot

sys.path.insert(0, '/usr/lib/python2.7/bridge/') 
from bridgeclient import BridgeClient as bridgeclient

bridge_client = bridgeclient()

"""
	This sample code demo button send to QIoT Suite 
	requirement:
	-- opkg update
	-- opkg install distribute
	-- opkg install python-openssl
	-- easy_install pip
	-- pip install paho-mqtt

    configure the system to allow various Bridge related services to run,command as follow 
	-- uci set yunbridge.config.disabled=’0’
	-- uci commit
	-- reboot
	run command: python button.py
"""

"""
	Setup connection options
"""
connection = None
connection = qiot.connection(qiot.protocol.MQTT)
connection_options = connection.read_resource('./res/resourceinfo.json', '/ssl/')
connection.connect(connection_options)

"""
	Send sensor's data to QIoT Suite Lite by Resourcetype.
"""
while True:	
	button_state = bridge_client.get("button")
	connection.publish_by_id("button", str(button_state))
	time.sleep(1)
